src/tezos/TezosOperations.ts

```diff
@@ -83,11 +83,22 @@ export namespace TezosOperations {
         network: string,
         blockHead: TezosTypes.BlockMetadata,
         operations: object[]): Promise<string> {
-        const payload = {
-                branch: blockHead.hash,
-                contents: operations
-            };
-        return TezosNode.forgeOperation(network, payload)
+        const payload = { branch: blockHead.hash, contents: operations };
+        const ops = await TezosNode.forgeOperation(network, payload);
+
+        const decoded = TezosMessageCodec.parseOperationGroup(ops);
+
+        for (let i = 0; i < operations.length; i++) {
```

Comment:
Is there a more canonical iteration we can use? I think a `for` loop like this is very rarely used in JS / TS.

Reply:
```
    operations.forEach(clientop => {
      decoded.forEach(serverop => {
        if (clientop["kind"] !== "transaction") {
          if (
            serverop.kind !== clientop["kind"] ||
            serverop.fee !== clientop["fee"] ||
            serverop.amount !== clientop["amount"] ||
            serverop.destination !== clientop["destination"]
          ) {
            return ops;
          }
        }
      });
    });
``` I think that'd do the trick